### ISSUE TITLE
App fails to load files when app path contains a # (hashtag)

### ISSUE THREAD

**Issue by vaughnroyko** [electron/electron#6869]
* Electron version: 1.3.3 (and previous versions as well, going back quite some time)
* Operating system: Windows 10 (could possibly happen on other operating systems as well)

No Electron errors seem to be produced. This happens with both directories and when using .asar packages. Simply load any app at all within a "#" folder to reproduce.

**Comment by baconface** [comment]
I would avoid a `#` in the path as a general practice. Doing this bad things will happen.  The `#` is treated differently for resource locations when used in a web app context. It basically going to get interpreted as a named anchor.

So `example#/resources/resource.js` would basically be treated as `example`.

**Comment by vaughnroyko** [comment]
I agree, but I have no control over folders/directories that exist on user's systems.

**Comment by zcbenz** [comment]
You have to use `url.format` to construct the URL passed to `loadURL` to solve this, there is no way on Electron's side to know whether the `#` in URL is intended.

We should also replace all `file://${__dirname}/index.html` with `url.format` calls in our documentation.

**Comment by zeke** [comment]
What's the best way to format a URL with the `file` protocol? Just setting `protocol` and `pathname` does not produce a good result:

```js
const path = require('path')
const url = require('url')

url.format({
  protocol: 'file', 
  pathname: path.join(__dirname, 'index.html')
})

// notice the missing slashes after the protocol
// file:Users/zeke/Desktop/index.html
```

**Comment by zeke** [comment]
Setting the `hostname` seems to work, but that just seems odd:

```
url.format({protocol: 'file', hostname: path.join(__dirname, 'index.html')})
// file:///Users/zeke/Desktop/index.html
```

**Comment by kevinsawicki** [comment]
> What's the best way to format a URL with the file protocol?

I think you want to set `slashes` to `true`.

https://nodejs.org/api/url.html#url_urlobject_slashes

**Comment by zcbenz** [comment]
Closed with #6900.
